### ISSUE TITLE
Question - Does TypeORM provide feature to do atomic increment with int field?

### ISSUE THREAD

**Issue by Diluka** [typeorm/typeorm#1080]
The shorthand of sql:
```sql
UPDATE table SET field = field + 1 WHERE id = :id;
```
Has there an API like
```ts
async increment(field:string,step:number=1);
```

**Comment by pleerock** [comment]
We'll need to implement this new feature. But first we need a design proposal on what other methods we also need (like increment, decrement, sum, etc.) - all common operations we'll need and create a separate type of query builder where those operations will be possible.

**Comment by pleerock** [comment]
@Diluka can you please help on proposal what other methods we can add as well?

```
increment
decrement
multiply // do we really need it?
divide // do we really need it?
sum
count
...
```

**Comment by pleerock** [comment]
Contributions are welcomed on this feature request btw.

**Comment by icepeng** [comment]
I think it would be nice to also support the date field.
Personally, multiply and divide methods do not seem to be necessary.
Is not sum and count a different category of functionality?

**Comment by daggett206** [comment]
@pleerock , hi, any chance to implement this until end of the year? :)

**Comment by pleerock** [comment]
@daggett206 no for sure, maybe we will implement it end of jan, if noone will do that. This feature is opened for contributors

**Comment by 9d77v** [comment]
#680

**Comment by pleerock** [comment]
Was implemented in #1845 and is available in latest `@next`. Example:

```ts
       // decrement counter of post 1
       await connection
           .getRepository(Post)
           .decrement({ id: 1 }, "counter", 1);

        // decrement counter of post 2
        await connection
            .manager
            .decrement(Post, { id: 2 }, "counter", 2);
```

**Comment by vincesp** [comment]
> Was implemented in #1845 and is available in latest `@next`. Example:
> 
> ```ts
>        // increment counter of post 1
>        await connection
>            .getRepository(Post)
>            .increment({ id: 1 }, "counter", 1);
> ```

How do I return the new value of the counter though? 🤔
